core/dbt/adapters/base/impl.py

```diff
@@ -1162,6 +1162,47 @@ def get_rows_different_sql(
 
         return sql
 
+    def valid_incremental_strategies(self):
+        return ["append", "delete+insert"]
+
+    @available.parse_none
+    def get_incremental_strategy_macro(self, strategy: str):
+        # Construct macro_name from strategy name
+        if strategy is None:
+            strategy = "default"
+
+        # validate strategies for this adapter
+        valid_strategies = self.valid_incremental_strategies()
+        valid_strategies.append("default")
+        builtin_strategies = ["append", "delete+insert", "merge", "insert_overwrite"]
+        if strategy in builtin_strategies and strategy not in valid_strategies:
```

Comment:
```python
if strategy in builtin_strategies
```
👍 I see, this is how we avoid validating custom strategies defined by _end users_, e.g. `silly` + `get_silly_incremental_sql`

What do you think about moving `builtin_strategies` into its own method? That way, if an external adapter plugin maintainer defines their own strategy (e.g. `special_athena_upsert`), they can update both methods (`valid_incremental_strategies` + `builtin_incremental_strategies`), and take advantage of this validation — without having to copy-paste-edit all of `get_incremental_strategy_macro` just to change one line.

Reply:
I've added a builtin_incremental_strategies method.